core/clustersmngr/clustersmngr.go

```diff
@@ -72,14 +74,19 @@ type clientsPool struct {
 	mutex   sync.Mutex
 }
 
-type ClusterClientConfig func(Cluster) *rest.Config
+type ClusterClientConfig func(Cluster) (*rest.Config, error)
 
 func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfig {
-	return func(cluster Cluster) *rest.Config {
+	return func(cluster Cluster) (*rest.Config, error) {
```

Comment:
It would be nice to have GoDoc for this `ClientConfigWithUser` function to explain what it does.